tests/Signals_perftest.m

```diff
@@ -1,90 +1,191 @@
 classdef Signals_perftest < matlab.perftest.TestCase
+% Runs performance tests for Signals
+%
+% Performs benchmarking for various operations on signals: 
+% nops, assignment ops, logical ops, scalar arithmetic ops, listener ops,
+% and signal methods: `map`, `scan`, `identity`, `subscriptable`, etc...
+% (via `test_mainOps` and `test_signalsOps` methods)
+%
+% And performs benchmarking for propogations through each signal in 
+% networks of various widths and depths (30 to 1000 nodes spread across 2
+% to 20 layers) (via `test_networkOps` method) (*Note: for reference, the 
+% `docs\examples\advancedChoiceWorld` exp def has 338 nodes over 10 layers)
+%
+% In this class, the 'Net' property is a Signals network, in which the 
+% 'A' & 'B' properties are treated as "input layer" signals from which all 
+% other signals are defined.
+%   
+% Example - display the mean measured time for running each test:
+%   results = runperf('Signals_perftest2.m');
+%   fullTableResults = vertcat(results.Samples);
+%   meanTimeByTestTable = varfun(@mean, fullTableResults,...
+%     'InputVariables', 'MeasuredTime', 'GroupingVariables', 'Name')
   
   properties
-    net
-    A
-    B
-    C
-    func
-    verify = false
-    results
-    reps = 2000
+    Net % Signals network ('sig.Net' object)
+    A % origin signal ('sig.node.OriginSignal' object)
+    B % origin signal ('sig.node.OriginSignal' object)
+    Reps = 1000 % number of test repetitions
+    MexOps % operations which are run directly in compiled mexnet
   end
   
   properties (TestParameter)
-    ops = num2cell([1:6 10:14 30 40 50])
+    % main operations to test (whose results are posted in the paper)
+    MainOps = {@post, @gt, @ge, @lt, @le, @eq, @plus, @minus, @times, ... 
```

Comment:
Basic and main still seems arbitrary to me.  `flattenStruct` is a MEX operation.  What makes `scan` a main op but not `at`? Related to this: we recently removed flattenStruct (https://github.com/cortex-lab/signals/pull/9/commits/0227241c2f1957a1289e0815f411beda56761061#diff-72b004a923af36db7da264b8291d1a84), however it is still referenced in the code which may be confusing: https://github.com/cortex-lab/signals/blob/500101aaf970400fca6656723a7faad348eee884/%2Bsig/%2Bnode/Signal.m#L293

Reply:
The differences between what functions are and aren't included in `MainOps` is arbitrary. This was always going to be the case, as we needed to pick some methods to use for benchmark results for the paper, and using all of them would have been overkill. This is mentioned in the doc in this file. I chose to use `scan` in `MainOps` because it is a more powerful and more commonly known function than `at`. 
Ok, I can remove `flattenStruct` from `sig.node.Signal`

Reply:
'main' suggests that those functions are in some way fundamental, more important, or more frequently used than the others, which doesn't strictly have anything to do with benchmarking. 

>> Ok, I can remove flattenStruct from sig.node.Signal

I'm not sure we want to remove flattenStruct from sig.node.Signal.  Maybe create an issue about it?  Ideally we follow the same design that map uses, but maybe have sig.transfer.flattenStruct contain only a docstring.  We should ask Chris first. Want to create an issue about this?

Reply:
Ok, I'll ask Chris about removing it, or keeping it as a docstring